Python Quickstarts (#25)
auth0/docs-v2: auth4genai/snippets/get-started/langchain-fastapi-py/async-auth.mdx

import { Prerequisites } from "/snippets/get-started/prerequisites/async-auth.jsx";

<Prerequisites
  callbackUrl="http://localhost:8000/api/auth/callback"
  logoutUrl="http://localhost:5173"
/>

### Prepare the FastAPI app

> **Recommended**: Use the starter template, clone the [Auth0 AI samples](https://github.com/auth0-samples/auth0-ai-samples) repository:

Reviewer: ```suggestion
**Recommended**: Use the start template by cloning [Auth0 AI samples](https://github.com/auth0-samples/auth0-ai-samples) repository:
```

```bash wrap lines
git clone https://github.com/auth0-samples/auth0-ai-samples.git
cd auth0-ai-samples/authenticate-users/langchain-fastapi-py
```

The project is divided into two parts:

- `backend/` contains the backend code for the Web app and API written in Python using FastAPI and the LangGraph agent.
- `frontend/` contains the frontend code for the Web app written in React as a Vite SPA.

### Install dependencies

In the `backend` directory of your project, install the following dependencies:

- `auth0-ai-langchain`: [Auth0 AI SDK for LangChain](https://github.com/auth0-lab/auth0-ai-python/tree/main/packages/auth0-ai-langchain) built for GenAI applications powered by LangChain.

> - `langgraph`: [LangGraph](https://pypi.org/project/langgraph/) for building stateful, multi-actor applications with LLMs.

Reviewer 2: multi-actor->multi-factor

- `langchain-openai`: LangChain integrations for OpenAI.
- `langgraph-cli`: LangGraph CLI for running a local LangGraph server.

Make sure you have [uv](https://docs.astral.sh/uv/) installed and run the following command to install the dependencies:

```bash wrap lines
cd backend
uv sync
uv add "auth0-ai-langchain>=1.0.0b3" "langgraph>=0.5.4" langchain-openai "langgraph-cli[inmem]" --prerelease=allow
```

### Update the environment file

> Copy the `.env.example` file to `.env` and update the variables with your Auth0 credentials. You can find your Auth0 domain, client ID and client secret in the application you created in the Auth0 dashboard.

Reviewer 2: comma after client ID
Auth0 dashboard->Auth0 Dashboard

Reviewer: ```suggestion
Copy the `.env.example` file to `.env` and update the variables with your Auth0 credentials. You can find your Auth0 domain, client ID, and client secret in the application you created in the Auth0 dashboard.
```

> ### Set up Human-in-the-Loop Approvals

Reviewer 2: Approvals->approvals

Reviewer: ```suggestion
### Set up Human-in-the-Loop approvals
```

> Integrate the Auth0 AI SDK into your application to secure your async AI agent workflow. For quickstart we will use a blocking request flow. In real use cases, often an asynchronous flow is preferred.

Reviewer 2: "For quickstart we will..."->"For this quickstart, we will..."
